lib/date_time/date_time_core.c

```diff
@@ -85,11 +116,14 @@ static void date_time_update_work_fn(struct k_work *work)
 	err = date_time_core_current_check();
 	if (err == 0) {
 		LOG_DBG("Using previously obtained time");
-		date_time_core_schedule_update(true);
+		date_time_core_schedule_update();
 		date_time_core_notify_event(DATE_TIME_EVT_TYPE_PREVIOUS);
 		return;
 	}
 
+	/* Clear the update_scheduled flag since the scheduled update has begun. */
+	atomic_clear(&update_scheduled);
```

Comment:
Comment is not true because this could be a manually triggered date_time update through `date_time_update_async`.